Tests/SotoCoreTests/Credential/ConfigFileCredentialProviderTests.swift

```diff
@@ -265,85 +190,78 @@ class ConfigFileCredentialProviderTests: XCTestCase {
+            let output = STSAssumeRoleResponse(credentials: stsCredentials)
             let xml = try XMLEncoder().encode(output)
             let byteBuffer = ByteBufferAllocator().buffer(string: xml.xmlString)
             let response = AWSTestServer.Response(httpStatus: .ok, headers: [:], body: byteBuffer)
             return .result(response)
-        })
-        var result: Credential?
-        XCTAssertNoThrow(result = try client.credentialProvider.getCredential(on: client.eventLoopGroup.next(), logger: AWSClient.loggingDisabled).wait())
-        let stsCredentials = result as? STSCredentials
-        XCTAssertEqual(stsCredentials?.accessKeyId, credentials.accessKeyId)
-        XCTAssertEqual(stsCredentials?.expiration, credentials.expiration)
-        XCTAssertEqual(stsCredentials?.secretAccessKey, credentials.secretAccessKey)
-        XCTAssertEqual(stsCredentials?.sessionToken, credentials.sessionToken)
-    }
-}
-
-// Extend STSAssumeRoleRequest so it can be used with the AWSTestServer
-extension STSAssumeRoleRequest: Decodable {
-    public init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
-        let roleArn = try container.decode(String.self, forKey: .roleArn)
-        let roleSessionName = try container.decode(String.self, forKey: .roleSessionName)
-        self.init(roleArn: roleArn, roleSessionName: roleSessionName)
-    }
-
-    private enum CodingKeys: String, CodingKey {
-        case roleArn = "RoleArn"
-        case roleSessionName = "RoleSessionName"
-    }
-}
-
-// Extend STSAssumeRoleResponse so it can be used with the AWSTestServer
-extension STSAssumeRoleResponse: Encodable {
-    public func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: CodingKeys.self)
-        try container.encode(credentials, forKey: .credentials)
-    }
+        }
 
-    private enum CodingKeys: String, CodingKey {
-        case credentials = "Credentials"
-    }
-}
+        let credentials = try futureCredentials.wait()
 
-// Extend STSCredentials so it can be used with the AWSTestServer
-extension STSCredentials: Encodable {
-    public func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: CodingKeys.self)
-        try container.encode(accessKeyId, forKey: .accessKeyId)
-        try container.encode(expiration, forKey: .expiration)
-        try container.encode(secretAccessKey, forKey: .secretAccessKey)
-        try container.encode(sessionToken, forKey: .sessionToken)
-    }
+        XCTAssertEqual(credentials.accessKeyId, stsCredentials.accessKeyId)
+        XCTAssertEqual(credentials.secretAccessKey, stsCredentials.secretAccessKey)
 
-    private enum CodingKeys: String, CodingKey {
-        case accessKeyId = "AccessKeyId"
-        case expiration = "Expiration"
-        case secretAccessKey = "SecretAccessKey"
-        case sessionToken = "SessionToken"
+        try testServer.stop()
```

Comment:
Can we use `defer { XCTAssertNoThrow(try ...` blocks here to keep it consistent with other tests